e2e/node_config.go

```diff
@@ -23,6 +23,87 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+// this is a base kubelet config for Scriptless e2e test
+var baseKubeletConfig = &aksnodeconfigv1.KubeletConfig{
+	EnableKubeletConfigFile: true,
+	KubeletFlags: map[string]string{
+		"--cloud-config":              "",
+		"--cloud-provider":            "external",
+		"--kubeconfig":                "/var/lib/kubelet/kubeconfig",
+		"--pod-infra-container-image": "mcr.microsoft.com/oss/kubernetes/pause:3.6",
+	},
+	KubeletNodeLabels: map[string]string{
+		"agentpool":                               "nodepool2",
+		"kubernetes.azure.com/agentpool":          "nodepool2",
+		"kubernetes.azure.com/cluster":            "test-cluster",
+		"kubernetes.azure.com/mode":               "system",
+		"kubernetes.azure.com/node-image-version": "AKSUbuntu-1804gen2containerd-2022.01.19",
+	},
+	KubeletConfigFileConfig: &aksnodeconfigv1.KubeletConfigFileConfig{
+		Kind:              "KubeletConfiguration",
+		ApiVersion:        "kubelet.config.k8s.io/v1beta1",
+		StaticPodPath:     "/etc/kubernetes/manifests",
+		Address:           "0.0.0.0",
+		TlsCertFile:       "/etc/kubernetes/certs/kubeletserver.crt",
+		TlsPrivateKeyFile: "/etc/kubernetes/certs/kubeletserver.key",
+		TlsCipherSuites: []string{
+			"TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
+			"TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
+			"TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305",
+			"TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
+			"TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305",
+			"TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
+			"TLS_RSA_WITH_AES_256_GCM_SHA384",
+			"TLS_RSA_WITH_AES_128_GCM_SHA256",
+		},
+		RotateCertificates: false,
```

Comment:
are we override `RotateCertificates` in the node config mutation? we definitely need this to be true for any e2e test to work since we rely on kubelet client TLS bootstrapping for e2e nodes to join the cluster

Reply:
Are you referring to this part of code?
https://github.com/Azure/AgentBaker/blob/devinwon/kubelet_config/parts/linux/cloud-init/artifacts/cse_config.sh#L438
${ENABLE_KUBELET_SERVING_CERTIFICATE_ROTATION} should be true, right?
There are 2 variables look similar to me. One is `--rotate-certificates`, `--rotate-server-certificates`. So I'd like to clarify first.

Reply:
--rotate-certificates allows the kubelet to use the bootstrap token (specified in bootstrap-kubeconfig) to bootstrap its _client_ certificate after startup, --rotate-server-certificates (named serverTLSBootstrap in kubelet config file) allows the kubelet to bootstrap its _serving_ certificate _after_ it has bootstrapped its client certificate - we always want to set --rotate-certificates to true, since currently AKS nodes rely on TLS bootstrapping to be able to join clusters (though this won't be the case once secure TLS bootstrapping is a thing...), though we don't necessarily always want to set --rotate-server-certificates to true, which is why ENABLE_KUBELET_SERVING_CERTIFICATE_ROTATION exists

Reply:
my original comment was just to make sure that in scriptless/aks-node-controller abe2e's that we're setting --rotate-certificates to true, otherwise the nodes would never be able to join the e2e cluster

Reply:
Thanks for clarifying. I did set "--rotate-certificates": "true" to the kubelet flag. But Let me also add it in Kubeletconfig to make sure they are aligned. Thanks for reminding me.